modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/table/Table.tsx

```diff
@@ -306,7 +306,7 @@ const EditFDSFieldModalContent = ({
 				<ClayForm.Group>
 					<ClayCheckbox
 						checked={fdsFieldSortable}
-						disabled={fdsField.type === EFieldType.OBJECT}
+						disabled={!fdsFieldSortable}
```

Comment:
won't this disable the checkbox if initially enabled and checked, then user unchecks it?

Reply:
Only if `sortable` value in DB is present before this change.

Reply:
Are you sure? :thinking: I think Dani might be right, this is a state variable changed in UI, few lines below this one.

Reply:
![image](https://github.com/liferay-frontend/liferay-portal/assets/132653342/1c11ecc5-0a95-4b8d-b209-90911a5f56e7)

Ooops, you are right.

Reply:
@markocikos @dsanz I did not realize that I was using the same property to handle different things :facepalm: 

The problem is that I was calculating the `sortable` property for each field in the `AddFieldsModalContent`, then saving it and storing the value in the DB, showing them in the `Table`.
However, when accessing the edition modal (subcomponent of `Table`), I only had the DB value and know nothing about the `sortable` properties calculated for the tree.

In order to fix this, I need to move some things from `AddFieldsModalContent` component and make them available from the `Table` component, so I could use in both components and compare the DB stored value with the tree calculated value.

Reply:
[Screencast from 2024-04-30 16-42-56.webm](https://github.com/liferay-frontend/liferay-portal/assets/132653342/164d9765-f34b-4272-95dd-6a117e8cf26f)

Reply:
Hmm I will end up negating most of these changes in my current task, that merges `AddFieldsModalContent` and `FieldSelectModalContent`. This is not just a simple case of conflicts, it's a divergence of logic that we are trying to avoid with the merger. The frustrating thing is I have a current blocker that prevents me from opening PR. I'll try to work through it asap.

Reply:
This is [my WIP branch](https://github.com/markocikos/liferay-portal/commit/7eb5125d0db209ee04399ec49b92fa7028b04dcd), so you can see in more detail what I mean. `AddFieldsModalContent` will no longer work in the same way, we're extracting all table-related logic, such as saving request, into `Table.tsx`, and this seems to me like it's negating or highly clashing with what you doing here. It may make sense to put this on hold until my task is done, unless I get bogged down in complexity.

Reply:
@markocikos Make sense. @herrardo FYI this PR will wait until https://github.com/liferay-frontend/liferay-portal/pull/4168 is merged